presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -501,52 +504,61 @@ public void testTimestampWithTimeZone()
                         .hasMessage(serverSupportsVariablePrecisionTimestampWithTimeZone()
                                 ? "Expected column to be a time type but is timestamp with time zone(3)" // TODO (https://github.com/prestosql/presto/issues/5317) placement of precision parameter
                                 : "Expected column to be a time type but is timestamp with time zone");
-                assertEquals(rs.getTimestamp(column), Timestamp.valueOf(LocalDateTime.of(2018, 2, 13, 6, 14, 15, 227_000_000))); // TODO this should fail, or represent TIMESTAMP '2018-02-13 13:14:15.227'
+                assertEquals(rs.getTimestamp(column), timestampForPointInTime);
             });
 
             // second fraction in nanoseconds overflowing to next second, minute, hour, day, month, year
             if (serverSupportsVariablePrecisionTimestampWithTimeZone()) {
                 checkRepresentation(connectedStatement.getStatement(), "TIMESTAMP '2019-12-31 23:59:59.999999999999 Europe/Warsaw'", Types.TIMESTAMP /* TODO TIMESTAMP_WITH_TIMEZONE */, (rs, column) -> {
-                    assertEquals(rs.getObject(column), Timestamp.valueOf(LocalDateTime.of(2019, 12, 31, 17, 0, 0, 0)));  // TODO this should represent TIMESTAMP TIMESTAMP '2019-12-31 23:59:59.999999999999 Europe/Warsaw'
+                    ZonedDateTime zonedDateTime = ZonedDateTime.of(2020, 1, 1, 0, 0, 0, 0, ZoneId.of("Europe/Warsaw"));
+                    Timestamp timestampForPointInTime = Timestamp.from(zonedDateTime.toInstant());
+                    assertEquals(rs.getObject(column), timestampForPointInTime);  // TODO this should represent TIMESTAMP TIMESTAMP '2019-12-31 23:59:59.999999999999 Europe/Warsaw'
```

Comment:
ditto

Reply:
Oh - here it actually makes sense, due to long fractional part.

Reply:
It's more about time zone actually.